indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -131,8 +137,12 @@ public KillUnusedSegmentsTask(
     if (limit != null && Boolean.TRUE.equals(markAsUnused)) {
       throw InvalidInput.exception("limit[%d] cannot be provided when markAsUnused is enabled.", limit);
     }
+    if (version != null && Boolean.TRUE.equals(markAsUnused)) {
+      throw InvalidInput.exception("version[%s] cannot be provided when markAsUnused is enabled.", version);
```

Comment:
I didn't add mark used/unused support for versions in this patch to keep the changes easy to review. Also, the `markAsUnused` option is deprecated, so maybe it's ok to keep it this way?

Reply:
think this is ok 👌